packages/@glimmer/helper/src/fn.ts

```diff
@@ -0,0 +1,11 @@
+import { DEBUG } from '@glimmer/env';
+import { assert } from '@glimmer/util';
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export default function fn(fn: (...args: unknown[]) => unknown, ...args: unknown[]) {
+  if (DEBUG) {
+    assert(typeof fn === 'function', 'fn must receive a function as its first parameter');
+    assert(args.length > 0, 'fn must receive at least one argument to pass to the function, otherwise there is no need to use fn.');
+  }
+  return fn.bind(null, ...args);
```

Comment:
We should use the "untouchable this" thing

https://github.com/emberjs/ember.js/blob/master/packages/@ember/-internals/glimmer/lib/utils/untouchable-this.ts

https://github.com/emberjs/ember.js/blob/master/packages/%40ember/-internals/glimmer/lib/helpers/fn.ts#L8